src/org/intellij/erlang/debugger/xdebug/ErlangXDebugProcess.java

```diff
@@ -124,17 +130,36 @@ public ErlangDebugLocationResolver getLocationResolver() {
   public synchronized void evaluateExpression(@NotNull String expression,
                                               @NotNull XDebuggerEvaluator.XEvaluationCallback callback,
                                               @NotNull ErlangTraceElement traceElement) {
-    // need to pause the debugging session otherwise the callback might get invalidated
-    mySession.pause();
     myEvalCallback = callback;
+    // see the comment about ignoreNextBreakpointReachedEvent
+    ignoreNextBreakpointReachedEvent = true;
     myDebuggerNode.evaluate(expression, traceElement);
   }
 
   @Override
   public synchronized void handleEvaluationResponse(OtpErlangObject response) {
     if (myEvalCallback != null) {
-      myEvalCallback.evaluated(ErlangXValueFactory.create(response));
-      mySession.resume();
+      String error = null;
```

Comment:
As for me, it's better to extract a separate function and assign it to the error variable.